web/packages/teleterm/src/ui/services/workspacesService/documentsService/types.ts

```diff
@@ -154,6 +166,7 @@ export interface DocumentPtySession extends DocumentBase {
 
 export type DocumentTerminal =
   | DocumentPtySession
+  | DocumentGatewayKube
   | DocumentGatewayCliClient
   | DocumentTshNode
   | DocumentTshKube;
```

Comment:
Documents are stored on disk in `app_state.json` – that's how we know what types you had open after you restart the app. After adding `DocumentGatewayKube`, we should be able to write a migration which rewrites existing `DocumentTshKube` to `DocumentGatewayKube`. Leave that migration to Grzegorz and I though, we should just remember to write it before backporting the PR with the kube gateway.

Reply:
Thanks. Speaking of backward compatibility, Connect should ship with the latest tsh right? Though I recall there is an option to use a different tsh?

Reply:
> Connect should ship with the latest tsh right?

Yes, Connect always ships with a tsh built from the same commit.

> Though I recall there is an option to use a different tsh?

Not really, there's `CONNECT_TSH_BIN_PATH` but it's for dev purposes only. The final packaged app always uses the tsh that is bundled with it.